growthbook/growthbook
# Send feature usage data to Clickhouse and show feature usage graphs.

### Features and Changes

- Sends feature-usage events to the ingestor
- Modifies the materialized view for events to not copy in feature-usage events
- Creates a materialized view for feature-usage events
- Creates backend code to query for feature usage
- Adds Feature Usage Graphs behind a feature flag

### Dependencies
Need to either destroy and recreate the growthbook datasource, or manually modify and create the feature-usage materialized view

### Testing
- yarn dev in growthbook-ingestor/packages/ingestor
- in .env.local set: `INGESTOR_HOST=http://localhost:3003`
- in .env.local set: `CLICKHOUSE_MAIN_TABLE=test_enriched_events`
- Create a feature flag with the same id as prod such as "bandits" that gets called when loading pages
- Go to that feature and see the feature usage graphs.

### Screenshots
![Screenshot 2024-12-18 at 6 31 56 PM](https://github.com/user-attachments/assets/1a408447-30ff-4c63-becc-05e3adb0db78)
![Screenshot 2024-12-18 at 6 31 44 PM](https://github.com/user-attachments/assets/79409c2a-7f08-496d-b057-88989259b671)
